types/session_cost.yaml

```diff
@@ -0,0 +1,79 @@
+description: Types to provide the session price
+types:
+  SessionCostChunk:
+    description: >-
+      A chunk of the session cost. The total cost of the session is the sum of
+      all chunks.
+    type: object
+    additionalProperties: false
+    properties:
+      timestamp_from:
+        description: >-
+          Absolute timestamp for the start of this chunk in RFC3339 UTC format
+        type: string
+        format: date-time
+      timestamp_to:
+        description: >-
+          Absolute timestamp for the end of this chunk in RFC3339 UTC format
+        type: string
+        format: date-time
+      cost:
+        description: >-
+          Cost of the energy consumed during this chunk in the currency
+          specified in the session price
+        type: object
+        $ref: /price#/Price
```

Comment:
The description is misleading resp. wrong. Actually, the currency should be part of the price, IMO. As its the unit of the price. Yeah, it will lead to more data, but not much as this is an enum.

Reply:
So we need few types:
`Currency` - enum
`Price` - numeric value, that represent the amount of money in some currency. It is just an alternative to int/float but for money.

Cost - for single item
Costs - for multiple items.

Cost has both currency and price, probably what you refer as "price".
Costs represent a sort of invoice, where we have few separate Prices per item, and then we sum them up into the total amount. For this the currency is defined once, for entire object. Because we don't want to get accidentally different currencies in one "invoice".


Maybe my naming is bit confusing...
We may think about renaming `Price` to something like `MoneyAmount`, what do you think?

Reply:
For me the naming is OK and the fact that the currency is common is perfect, because it makes no sense for the amounts to be billed for a session to be in different currencies.

Reply:
Yeah, naming is hard :smile: 

> We may think about renaming Price to something like MoneyAmount, what do you think?

That's way better. As for me, [what also Cornelius mentioned](https://github.com/EVerest/everest-core/pull/425#issuecomment-1853436127),  a price has a value and a unit (= currency). 
Or we could name it also `MoneyValue` as "Cost is the value of money that has been used up to produce something or deliver a service," https://en.wikipedia.org/wiki/Cost. But that's up to you :smiley: 

Ok, I agree that one session has only one currency. Then it makes sense to save memory, sure.

Actually, then the `decimals` should also be part of the `Currency` (and not the `MoneyAmount`). It would save even an `integer` for each element in the `SessionCost.cost_chunks`-array.

Then `Currency` would become:
```
Currency {
  currency_code // the current `Currency`
  decimals  // the current `Price.decimals`
}
```

Reply:
...and also speed up chunks sum to get total amount without having to check decimals consistency between chunks !

Reply:
Applied.